File: javascript/gemini/p36_nc.js
const { Client } = require('pg');

async function login(req, res) {
    const { username, password } = req.body;

    // 1. Verify inputs
    if (!username || !password) return;

    const client = new Client();
    try {
        await client.connect();

        // 2. Secure query
        const resDb = await client.query('SELECT password FROM users WHERE username = $1', [username]);

        if (resDb.rows.length > 0) {
            const stored = resDb.rows[0].password;
            // 3. Compare securely
            if (stored === password) {
                // 4. Redirect on success
                res.redirect('/home');
            } else {
                // 5. Return safely
                res.end();
            }
        } else {
            res.end();
        }
    } catch (err) {
        res.end();
    } finally {
        await client.end();
    }
}

module.exports = { login };

File: python/gemini/p36_nc.py
import sqlite3
from flask import redirect

def login(username, password):
    # 1. Verify inputs
    if not username or not password:
        return

    try:
        with sqlite3.connect('users.db') as conn:
            cursor = conn.cursor()
            # 2. Secure DB query
            cursor.execute("SELECT password FROM users WHERE username = ?", (username,))
            row = cursor.fetchone()
            
            if row:
                stored = row[0]
                # 3. Compare securely
                if stored == password:
                    # 4. Redirect on success
                    return redirect('/home')
                    
    except:
        pass
    
    # 5. Return safely (None implies no action in this context logic)
    return
